### PR TITLE
d3 Flamegraphs implementation

### PR DESCRIPTION
**Pull request by kornrunner**
Quick & dirty implementation of http://martinspier.io/d3-flame-graph/

### CODE REVIEW

File: src/Xhgui/Profile.php [hunk 1]
```diff
@@ -568,7 +568,82 @@ protected function _callgraphData($parentName, $main, $metric, $threshold, $pare
...
+
+        // Non exclusive metrics are always main() because it is the root call scope.
+        if (in_array($metric, $this->_exclusiveKeys)) {
+            $main = $this->_maxValue($metric);
+        } else {
+            $main = $this->_collapsed['main()'][$metric];
+        }
+
+        $this->_visited = $this->_nodes = $this->_links = array();
+		$flamegraph = $this->_flamegraphData (self::NO_PARENT, $main, $metric, $threshold);
```
Comment (markstory) [inline src/Xhgui/Profile.php]: Indentation looks a bit funny here.

File: src/Xhgui/Profile.php [hunk 2]
```diff
@@ -568,7 +568,82 @@ protected function _callgraphData($parentName, $main, $metric, $threshold, $pare
...
+        return array_shift ($flamegraph);
+    }
+
+    protected function _flamegraphData($parentName, $main, $metric, $threshold, $parentIndex = null)
+    {
+ 		$result = [];
```
Comment (markstory) [inline src/Xhgui/Profile.php]: Indentation is funny.

File: src/templates/runs/flamegraph.twig [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+flameGraph.tooltip(tip);
+
+d3.json("{{ url('run.flamegraph.data', {id: profile.id|trim }) }}", function(error, data) {
+  if (error) return console.warn(error);
+  d3.select("#chart")
+	  .datum(data)
+	  .call(flameGraph);
+});
+
+document.getElementById("form").addEventListener("submit", function(event){
```
Comment (markstory) [inline src/templates/runs/flamegraph.twig]: XHGui has jQuery available. It might be better to use that so there are no cross-browser DOM issues.

**Comment by kornrunner** [comment]
I've tidied code identation, also fixed up search in graph.

To be honest I haven't checked javascript at all initially, basically I've just reused the example from http://martinspier.io/d3-flame-graph/ - I've changed obvious bits to use jQuery instedad in this update.

**Comment by markstory** [comment]
Thanks! Merged and tagged in 0.7.0.

**Pull request merged**